client/state/memberships/subscriptions/reducer.js

```diff
@@ -0,0 +1,44 @@
+/** @format */
+
+/**
+ * Internal dependencies
+ */
+
+import {
+	MEMBERSHIPS_SUBSCRIPTIONS_LIST,
+	MEMBERSHIPS_SUBSCRIPTIONS_RECEIVE,
+	MEMBERSHIPS_SUBSCRIPTIONS_LIST_SUCCESS,
+	MEMBERSHIPS_SUBSCRIPTIONS_LIST_FAILURE,
+} from 'state/action-types';
+import { combineReducers, createReducer } from 'state/utils';
+
+/**
+ * Returns the updated items state after an action has been dispatched.
+ * The state contains all past and upcoming billing transactions.
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ */
+export const items = createReducer( [], {
+	[ MEMBERSHIPS_SUBSCRIPTIONS_RECEIVE ]: ( state, { subscriptions } ) => subscriptions,
+} );
```

Comment:
there's no need to overcomplicate this here - why not use a basic reducer instead?

```js
export const items = ( state = [], { type, subscriptions } ) =>
	type === MEMBERSHIPS_SUBSCRIPTIONS_RECEIVE
		? subscriptions
		: state;
```

again…this is a perfect candidate for "shove data" - let's talk about `httpData()` for it